pest/src/span.rs

```diff
@@ -62,6 +62,32 @@ impl<'i> Span<'i> {
         }
     }
 
+    /// Attempts to create a new span based on a sub-range.
+    ///
+    /// TODO better docs
+    pub fn sub_span(&self, range: impl std::ops::RangeBounds<usize>) -> Option<Span<'i>> {
+        let start = match range.start_bound() {
+            std::ops::Bound::Included(&offset) => offset,
+            std::ops::Bound::Excluded(&offset) => offset + 1,
+            std::ops::Bound::Unbounded => 0,
+        };
+        let end = match range.end_bound() {
+            std::ops::Bound::Included(&offset) => Some(offset + 1),
+            std::ops::Bound::Excluded(&offset) => Some(offset),
+            std::ops::Bound::Unbounded => None,
+        };
+        let s = self.as_str();
+        if s.get(start..end.unwrap_or(s.len())).is_some() {
```

Comment:
@Alextopher one more item to fix: the merge conflict seems to be about `core` vs `std`. pest now supports `no_std`, so these `ops` may either need to be adjusted for `no_std` or sub_span should be feature-guarded for `std`